src/ray/core_worker/reference_count.cc

```diff
@@ -483,6 +483,7 @@ int64_t ReferenceCounter::ReleaseLineageReferences(ReferenceTable::iterator ref)
       RAY_CHECK(arg_it->second.on_ref_removed == nullptr);
       lineage_bytes_evicted += ReleaseLineageReferences(arg_it);
       EraseReference(arg_it);
+      ReleasePlasmaObject(arg_it);
```

Comment:
The fix looks good to me! I think we just need to move this to go before the above line, though, because the previous line deletes `arg_it`.

Reply:
Actually I think you could also just call DeleteReferenceInternal instead of EraseReference and ReleasePlasmaObject.